astropy_helpers/commands/build_sphinx.py

```diff
@@ -170,21 +172,22 @@ def run(self):
         # prevents a continuous updating at the terminal, but there's no
         # apparent way around this.
         if self.warnings_returncode:
-            proc = subprocess.Popen([sys.executable],
+            proc = subprocess.Popen([sys.executable, '-c', subproccode.encode('utf-8')],
```

Comment:
You're mixing bytes and strings here.  On Python 3 `sys.executable` and `'-c'` are strings, while the result of `.encode('utf-8')` is a bytes object.  In this case you don't need the `.encode('utf-8')`, I think.